cmd/podman/oci-trace-hook/trace.go

```diff
@@ -0,0 +1,238 @@
+package main
+
+import (
+	"bufio"
+	"bytes"
+	"encoding/binary"
+	"encoding/json"
+	"flag"
+	"io/ioutil"
+	"log"
+	"os"
+	"os/signal"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/docker/docker/api/types"
+	bpf "github.com/iovisor/gobpf/bcc"
+	spec "github.com/opencontainers/runtime-spec/specs-go"
+	seccomp "github.com/seccomp/libseccomp-golang"
+)
+
+type event struct {
+	Pid uint32
+	ID  uint32
+	// Inum    uint
+	Command [16]byte
+}
+
+type calls map[string]int
+
+const source string = `
+#include <linux/bpf.h>
+#include <linux/nsproxy.h>
+#include <linux/pid_namespace.h>
+#include <linux/ns_common.h>
+#include <linux/sched.h>
+#include <linux/tracepoint.h>
+
+BPF_HASH(parent_namespace, u64, unsigned int);
+BPF_PERF_OUTPUT(events);
+
+struct data_t
```

Comment:
Is this a C version of the Go event struct? We shouldn't have two versions of the same struct... It's possible to refer to C structures in Go code, so we should be using the C version everywhere.

Reply:
Yes, that's essentially the same and agree to use the C type. When reading the data from the bpf channel, it should be possible to: `s/var e event/var e C.data_t` (and maybe rename `data_t` to something more expressive).